src/utlist.h

```diff
@@ -600,6 +620,31 @@ do {
   }                                                                                            \
 } while (0)
 
+#define DL_INSERT_INORDER(head,add,cmp)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    DL_FIND_POS(_pos,add,cmp);                                                                 \
+    DL_APPEND_ELEM(head,_pos,add);                                                             \
+  } while (0)
+
+
+#define DL_FIND_POS(head,node,cmp)                                                             \
+  DL_FIND_POS2(head,node,cmp,prev,next)                                                        \
+
+#define DL_FIND_POS2(head,node,cmp,prev,next)                                                  \
+  do {                                                                                         \
+    if (cmp((head), (node)) > 0) {                                                             \
```

Comment:
What if `head` is `NULL`? (And please add a test case that covers that possibility.)